.github/workflows/test.yml

```diff
@@ -0,0 +1,23 @@
+name: Run tests
```

Comment:
can we rename the workflow to be something like **CI/static_analysis {matrix_info}**

Reply:
I'm not sure we should include the matrix information in the workflow name. Let me know if my solution works for you.

Reply:
well we don't have to, GH Actions automatically append matrix info to job names